File: nysol_python/mcmd/methods/convTex/db/smp/mhashsum.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mhashsum'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer,quantity,amount
A,1,
B,,15
A,2,20
B,3,10
B,1,20
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``customer`` 項目を単位にして、 ``quantity`` と ``amount`` 項目の合計を計算する。
'''
script['sh_code']='''
mhashsum k=顧客 f=数量,金額 i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mhashsum(k="customer", f="quantity,amount", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='NULL値出力'
script['comment']='''
``n=True`` オプションを指定することで、NULL値が含まれている場合は、結果もNULL値として出力する。
'''
script['sh_code']='''
mhashsum k=顧客 f=数量,金額 -n i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mhashsum(k="customer", f="quantity,amount", n=True, i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mhashsum.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer,Quantity,Amount
A,1,
B,,15
A,2,20
B,3,10
B,1,20
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Calculate the total \verb|Quantity| and total \verb|Amount| for each \verb|Customer| using the hash function.
EOF
scp=<<'EOF'
more dat1.csv
mhashsum k=Customer f=Quantity,Amount i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="NULL value in output"
comment=<<'EOF'
The output returns NULL if NULL value is present in \verb|Quantity| and \verb|Amount|. Use \verb|-n| option to print the null value.
EOF
scp=<<'EOF'
mhashsum k=Customer f=Quantity,Amount -n i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)
